bpf/Makefile

```diff
@@ -38,36 +49,55 @@ space := ${null} ${null}
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1 \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_UDP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_TCP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_ENCAP_HOST_REMAP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_UDP:-DENABLE_NODEPORT=1 \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1 \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1 \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION \
+	-DENABLE_IPV4=1:-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY \
+	-DENABLE_IPV4=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY \
+	-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY \
+	-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY:-DENABLE_SRC_RANGE_CHECK \
+	-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY:-DENABLE_BANDWIDTH_MANAGER \
+	-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY:-DENABLE_BANDWIDTH_MANAGER:-DENABLE_SRC_RANGE_CHECK \
+	-DENABLE_IPV6=1:-DENCAP_IFINDEX=1:-DENABLE_IPSEC:-DENABLE_HOST_SERVICES_UDP:-DENABLE_HOST_SERVICES_TCP:-DENABLE_NODEPORT=1:-DENABLE_EXTERNAL_IP:-DENABLE_NODEPORT_ACCELERATION:-DENABLE_SESSION_AFFINITY:-DENABLE_BANDWIDTH_MANAGER:-DENABLE_SRC_RANGE_CHECK:-DLB_SELECTION:-DLB_SELECTION_MAGLEV
```

Comment:
Is it possible to generate this combinations instead of listing them?

Reply:
That's the longer term goal, but definitely out of scope for this PR. To generate this, we will have to parse the macros from our bpf_xxx.c and .h files and detect impossible and uninteresting combinations. The long term effort toward this is tracked in the context of SIG Datapath, including by #5058.